utility/scriptConfig.py

```diff
@@ -45,22 +47,60 @@ class BaseScriptConfig:
     for scripts which can be run from the admin UI.
     """
 
-    def __init__(self, help: str, args: t.Optional[t.List[Arg]], arglist: t.Optional[t.List[str]]):
-        parser = argparse.ArgumentParser(description=help)
+    def __init__(
+        self,
+        help: str,
+        args: t.Optional[list[Arg]],
+        arglist: t.Optional[list[str]],
+        mutually_exclusive_args: t.Optional[tuple[tuple[Arg]]] = None,
+    ):
+        self.parser = argparse.ArgumentParser(description=help)
         self.help: Help = {'help': help, 'options': {}}
         self.args_dict = {}
+        self.mutually_exclusive_arg_flags = set()
+        if mutually_exclusive_args:
+            self._add_mutually_exclusive_args(mutually_exclusive_args)
         if args:
-            for arg in args:
+            self._add_args(args)
+        self.args = self.parser.parse_args(arglist)
+        self.defaults = [self.parser.get_default(key) for key in self.args.__dict__.keys()]
+        
+    def _add_mutually_exclusive_args(self, mutually_exclusive_args: tuple[tuple[Arg]]):
+        for mutually_exclusive_args_list in mutually_exclusive_args:
+            group = self.parser.add_mutually_exclusive_group()
+            for arg in mutually_exclusive_args_list:
                 flag = arg.pop('flag')
-                parser.add_argument(flag, **arg)
-                arg_name = flag.lstrip('-').replace('-', '_')
-                # some args with the 'store_true' action do not specify a type
-                # NOTE: maybe we should just specify it everywhere?
-                self.args_dict[arg_name] = {'type': type(arg['default']).__name__, 'default': arg['default']}
-                if 'help' in arg:
-                    self.help['options'][arg_name] = arg['help']
-        self.args = parser.parse_args(arglist)
-        self.defaults = [parser.get_default(key) for key in self.args.__dict__.keys()]
+                group.add_argument(flag, **arg)
+                self.mutually_exclusive_arg_flags.add(flag)
+                self._add_argument_to_args_dict(flag, arg)
+            group_arg_names = tuple(map(lambda arg: arg.dest, group._group_actions))
+            for arg_name in group_arg_names:
+                self._update_argument_in_args_dict(
+                    arg_name,
+                    mutually_exclusive_with=[
+                        another_arg_name for another_arg_name in group_arg_names
+                        if another_arg_name != arg_name
+                    ]
+                )
+                
+    def _add_args(self, args: list[Arg]):
+        for arg in args:
+            flag = arg.pop('flag')
+            if flag in self.mutually_exclusive_arg_flags:
+                continue
+            self.parser.add_argument(flag, **arg)
+            self._add_argument_to_args_dict(flag, arg)
+        
+    def _add_argument_to_args_dict(self, flag: str, arg: Arg):
```

Comment:
It might be better to accept the `arg_name` instead of the flag, the same way `_update_argument_in_args_dict()` does. You should be able to get the `.dest` from the return value of `parser.add_argument()`.